carsus/io/output/base.py

```diff
@@ -486,42 +544,55 @@ def _create_levels_lines(self, lines_loggf_threshold=-3,
         levels = levels.sort_values(
             ["atomic_number", "ion_number", "level_number"])
 
-        self.lines = lines
-        self.levels = levels
+        return levels, lines
 
-    def _create_collisions(self):
-        # Exclude artificially created levels from levels
-        levels = self.levels.loc[self.levels["level_id"] != -1].set_index("level_id")
+    def create_collisions(self, temperatures=np.arange(2000, 50000, 2000)):
+        """
+        Returns the same output than `AtomData.create_collisions` method.
 
-        ch_list = []
-        ch = self.chianti_reader
+        Generates the definitive `collisions` DataFrame by adding new columns
+        and making some calculations.
 
-        logger.info('Ingesting collisions from Chianti')
-        for ion in self.chianti_ions:
+        """
 
-            df = ch.collisions.loc[ion]
-            df = self.get_lvl_index2id(df, self.levels_all, ion)
-            ch_list.append(df)
+        logger.info('Ingesting collisional strengths.')
+        ch_collisions = self.chianti_reader.collisions
+        ch_collisions['ds_id'] = 4
 
-        collisions = pd.concat(ch_list, sort=True)
-        collisions['source'] = 'chianti'
+        # Not really needed because we have only one source of collisions
+        collisions = pd.concat([ch_collisions], sort=True)
+        ions = self.chianti_ions
+
+        collisions = collisions.reset_index()
+        collisions = collisions.rename(columns={'ion_charge': 'ion_number'})
+        collisions = collisions.set_index(['atomic_number', 'ion_number'])
 
-        # Keep this value to compare against SQL
-        collisions['ds_id'] = 4
+        logger.info('Matching collisions and levels.')
+        col_list = [ self.get_lvl_index2id(collisions.loc[ion], self.levels_all)
+                        for ion in ions]
+        collisions = pd.concat(col_list, sort=True)
+        collisions = collisions.sort_values(by=['lower_level_id', 'upper_level_id'])
 
         # `e_col_id` number starts after the last line id
         start = self.lines_all.index[-1] + 1
         collisions['e_col_id'] = range(start, start + len(collisions))
-        collisions = collisions.reset_index()
+
+        # Exclude artificially created levels from levels
+        levels = self.levels.loc[self.levels["level_id"] != -1].set_index("level_id")
```

Comment:
what does -1 stand for - why are they -1.

Reply:
Artificial levels, I think you wrote that.

Reply:
```
        # Reset indexes because `level_id` cannot be an index once we
        # add artificial levels for fully ionized ions that don't have ids (-1)
```

Reply:
Add a comment here that artificial levels for fully ionized ions that don't have ids have level_id = -1.